deps/oculus/src/ovrsession.cpp

```diff
@@ -1,14 +1,46 @@
-#include <stdlib.h>
+#include <deque>
 #include <iostream>
 #include <ovrsession.h>
+#include <stdlib.h>
+#include <mutex>
 #include <node.h>
+#include <thread>
 #include <v8.h>
 
 #include <Extras/OVR_Math.h>
 
 using namespace v8;
 using namespace OVR;
 
+namespace oculusvr {
+  uv_sem_t reqSem;
+  uv_async_t resAsync;
+  std::mutex reqMutex;
+  std::mutex resMutex;
+  std::deque<std::function<void()>> reqCbs;
+  std::deque<std::function<void()>> resCbs;
+  std::thread reqThead;
+
+  void RunResInMainThread(uv_async_t *handle) {
+    Nan::HandleScope scope;
+
+    std::function<void()> resCb;
+    {
+      std::lock_guard<std::mutex> lock(reqMutex);
```

Comment:
Should this be `resMutex`? It could result in corruption if we're not locking the right thing.